core/providers/rust/rust.go

```diff
@@ -0,0 +1,693 @@
+	}
+}
+
+func (p *RustProvider) InstallMisePackages(ctx *generate.GenerateContext, miseStep *generate.MiseStepBuilder) {
+	rust := miseStep.Default("rust", DEFAULT_RUST_VERSION)
+
+	cargoToml, _ := parseCargoTOML(ctx)
+	if cargoToml != nil {
+		// Fall back to the edition
+		switch cargoToml.Package.Edition {
+		case "2015":
+			// https://doc.rust-lang.org/edition-guide/rust-2015/index.html
+			// >= 1.0.0
+			miseStep.Version(rust, "1.30.0", "Cargo.toml")
+		case "2018":
+			// https://doc.rust-lang.org/edition-guide/rust-2021/index.html
+			// >= 1.31.0
+			miseStep.Version(rust, "1.55.0", "Cargo.toml")
+		case "2021":
+			// https://doc.rust-lang.org/edition-guide/rust-2021/index.html
+			// >= 1.56.0
+			miseStep.Version(rust, "1.84.0", "Cargo.toml")
+		case "2024":
+			// https://doc.rust-lang.org/edition-guide/rust-2024/index.html
+			// >= 1.85.0
+			miseStep.Version(rust, "1.85.1", "Cargo.toml")
+		}
+	}
+
+	if envVersion, varName := ctx.Env.GetConfigVariable("RUST_VERSION"); envVersion != "" {
+		miseStep.Version(rust, envVersion, varName)
+	}
+
+	for _, filename := range []string{"rust-version.txt", ".rust-version"} {
+		if content, err := ctx.App.ReadFile(filename); err == nil {
+			if version := strings.TrimSpace(utils.ExtractSemverVersion(content)); version != "" {
+				miseStep.Version(rust, version, filename)
+			}
+		}
+	}
+
+	if cargoToml != nil {
+		if cargoToml.Package.RustVersion != "" {
+			// Newer versions of Rust allow the `rust-version` field in Cargo.toml
+			if version := utils.ExtractSemverVersion(cargoToml.Package.RustVersion); version != "" {
+				miseStep.Version(rust, version, "Cargo.toml")
+			}
+		}
+	}
+
+	if toolchain, err := parseRustToolchain(ctx); err == nil {
+		if version := utils.ExtractSemverVersion(toolchain.Toolchain.Version); version != "" {
+			miseStep.Version(rust, version, "rust-toolchain.toml")
+		}
+	}
+}
+
+func (p *RustProvider) shouldUseMusl(ctx *generate.GenerateContext) bool {
+	if p.shouldMakeWasm32Wasi(ctx) {
+		return false
+	}
+
+	if ctx.Env.IsConfigVariableTruthy("NO_MUSL") {
+		return false
+	}
+
+	toolchainFile, _ := parseRustToolchain(ctx)
+	if toolchainFile != nil {
+		return false
+	}
+
+	if p.usesOpenssl(ctx) {
+		return false
+	}
+
+	return true
+}
+
+var wasmRegex = regexp.MustCompile(`target\s*=\s*"wasm32-wasi"`)
+
+func (p *RustProvider) shouldMakeWasm32Wasi(ctx *generate.GenerateContext) bool {
+	matches := ctx.App.FindFilesWithContent(".cargo/config.toml", wasmRegex)
+	return len(matches) > 0
+}
+
+func (p *RustProvider) usesOpenssl(ctx *generate.GenerateContext) bool {
+	app := ctx.App
+	// Check Cargo.toml
+	cargoToml, err := parseCargoTOML(ctx)
+	if err == nil {
+		// Check for "openssl" in any of the dependency maps
+		if _, ok := cargoToml.Dependencies["openssl"]; ok {
+			return true
+		}
+		if _, ok := cargoToml.DevDependencies["openssl"]; ok {
+			return true
+		}
+		if _, ok := cargoToml.BuildDependencies["openssl"]; ok {
+			return true
+		}
```

Comment:
Extract to `usesDep` function? I execpt it will be used for other checks too

Reply:
I don't think it's necessary if we don't do musl : )